lib/cmock_header_parser.rb

```diff
@@ -43,12 +45,55 @@ def parse(name, source)
 
     { :includes  => nil,
       :functions => @funcs,
-      :typedefs  => @typedefs
+      :typedefs  => @typedefs,
+      :normalized_source    => transform_inline_functions(source),
     }
   end
 
   private if $ThisIsOnlyATest.nil? ################
 
+  def remove_nested_pairs_of_braces(source)
+    # remove nested pairs of braces because no function declarations will be inside of them (leave outer pair for function definition detection)
+    if (RUBY_VERSION.split('.')[0].to_i > 1)
+      #we assign a string first because (no joke) if Ruby 1.9.3 sees this line as a regex, it will crash.
+      r = "\\{([^\\{\\}]*|\\g<0>)*\\}"
+      source.gsub!(/#{r}/m, '{ }')
+    else
+      while source.gsub!(/\{[^\{\}]*\{[^\{\}]*\}[^\{\}]*\}/m, '{ }')
+      end
+    end
+
+    return source
+  end
+
+  def transform_inline_functions(source)
+    # let's clean up the encoding in case they've done anything weird with the characters we might find
+    source = source.force_encoding("ISO-8859-1").encode("utf-8", :replace => nil)
+
+    source.gsub!(/(static|inline)+.*\{.*\w*\}/m) do |m|
+      m.gsub!(/(static|inline)/, '') # remove static and inline keywords
+      m = remove_nested_pairs_of_braces(m)
+
+      # Functions having "{ }" at this point are/were inline functions,
+      # Disguise them as normal functions with the ";"
+      m.gsub!(/\s*\{\s\}/, ";")
+
+      # Cleanup the function declarations
+      # Not strictly necessary, it will compile just fine, but it can help during debugging
+      m_lines = m.split(/\s*;\s*/).uniq
+      m_lines.each {
+        |m_line|
+        m_line.gsub!(/^\s+/, '')         # remove extra white space from beginning of line
+        m_line.gsub(/\s+/, ' ')          # remove remaining extra white space
```

Comment:
For these lines, you want `gsub!` not `gsub`, otherwise it's not changing `m_line`, it's just returning the result.

Reply:
Ah yes, thanks, good catch.